math/fibonacci.c

```diff
@@ -1,23 +1,40 @@
+/**
+ * @file fibonacci.c
+ * @brief Program to print the nth term of the Fibonacci series.
+ * @details Fibonacci series generally starts from 0 and 1. Every next term in
+ *the series is equal to the sum of the two preceding terms.
+ **/
+
 #include <stdio.h>
 
-// Fibonnacci function
-int fib(int number)
-{
-    if (number == 1 || number == 2)
-        return 1;
-    else
-        return fib(number - 1) + fib(number - 2);
-}
+unsigned fib(int);
 
+/* Driver Code */
 int main()
 {
     int number;
 
-    // Asks for the number that is in n position in Fibonnacci sequence
-    printf("Number: ");
+    // Getting n
+    printf("Enter n to find nth fibonacci element: ");
     scanf("%d", &number);
 
-    printf("%d \n", fib(number));
-
+    printf("%u\n", fib(number));
     return 0;
-}
\ No newline at end of file
+}
+
+/**
+ * @brief Determines the nth Fibonacci term
+ * @param number - n in "nth term" and it can't be negative
+ * @return nth term in unsigned type
+ * @warning Only till 47th and 48th fibonacci element can be stored in 'int' and
+ * 'unsigned' respectively (takes more than 20 seconds to print)
+ */
+unsigned fib(int number)
+{
+    if (number == 1)
+        return 0;
+    else if (number == 2)
```

Comment:
Why do you change the value in 1st number value?

If that, what if to avoid `else`?
something like:
```c
if (number == 1)
    return 0;

if (number == 2)
    return 1;

return fib(number - 1) + fib(number - 2);
```

Reply:
[Wikipedia](https://en.wikipedia.org/wiki/Fibonacci_number), the current definition starts the sequence from `0`

Reply:
I see there are many ways where to start from.

Reply:
I suppose that's also true, 🤔